src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs

```diff
@@ -270,5 +276,29 @@ private void VerifyAuthorizationResult()
                 throw new MsalServiceException(_authorizationResult.Error, _authorizationResult.ErrorDescription, null);
             }
         }
+
+        private async Task CheckForBrokerAndAcquireAuthorizationAsync(CancellationToken cancellationToken)
+        {
+            Broker = brokerFactory.CreateBrokerFacade(ServiceBundle.DefaultLogger);
+
+            if (Broker.CanInvokeBroker(_interactiveParameters.UiParent, ServiceBundle))
+            {
+                _msalTokenResponse = await Broker.AcquireTokenUsingBrokerAsync(
+                    CreateBrokerPayload(),
+                    ServiceBundle).ConfigureAwait(false);
+            }
+            else
+            {
+                await AcquireAuthorizationAsync().ConfigureAwait(false);
+                VerifyAuthorizationResult();
+                _msalTokenResponse = await SendTokenRequestAsync(GetBodyParameters(), cancellationToken).ConfigureAwait(false);
+            }
```

Comment:
should we throw an exception instead?

Reply:
let's discuss w/the other issue.

---
In reply to: [256611519](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/881#discussion_r256611519) [](ancestors = 256611519)

Reply:
we didn't discuss this. #todo.

---
In reply to: [256683730](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/881#discussion_r256683730) [](ancestors = 256683730,256611519)